dispatch/dispatch.go

```diff
@@ -161,20 +167,44 @@ func (d *Dispatcher) run(it provider.AlertIterator) {
 				return
 			}
 
-			level.Debug(d.logger).Log("msg", "Received alert", "alert", alert)
+			// this block is wrapped in a function to make sure that the span
+			// is ended before the next alert is processed
+			func() {
+				traceCtx, span := tracer.Start(d.ctx, "dispatch.Dispatcher.dispatch",
+					trace.WithAttributes(
+						attribute.String("alert.name", alert.Name()),
```

Comment:
Compared to the fingerprint, `alert.Name()` is not used much in Alertmanager. We might be able to omit it here as it's a lot of data, and I think most operators have been fine using the fingerprint in the past.

Reply:
> We might be able to omit it here as it's a lot of data, and I think most operators have been fine using the fingerprint in the past.

isn't it what gets sent as the `alertname` label? surely that's a lot more human-readable than the fingerprint?

Reply:
You could have thousands of alerts all with the same alertname. That's where fingerprint is useful, as it is a unique identifier for an alert.

Reply:
From an operational perspective I think the alert name is still quite valuable... Here's a sample from my test setup:

```
alert.fingerprint | "2a7a97fbf890646a"
alert.name | "AlwaysFiring"
```

I wouldn't be able to relate this trace back to my alert configuration without seeing `AlwaysFiring`...

Reply:
I think as long as we have both then 👍